ERROR: cannot find function `find` in module `strings`

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```rust
fn testcsv() -> Result<(), Box<dyn Error>> {
    let q = LazyCsvReader::new("000338.csv")
        .has_header(true)
        .finish()?
        .select([col("datetime").str().to_datetime(
            Some(TimeUnit::Microseconds),
            None,
            StrptimeOptions::default(),
            lit("raise"),
        )])
        .collect()?;

    dbg!(q);

    Ok(())
}
```

### Log output

```shell
Compiling polars-plan v0.37.0
error[E0425]: cannot find function `find` in module `strings`
   --> D:\App\Rust\cargo\registry\src\index.crates.io-6f17d22bba15001f\polars-plan-0.37.0\src\dsl\function_expr\strings.rs:292:64
    |
292 |             Find { literal, strict } => map_as_slice!(strings::find, literal, strict),
    |                                                                ^^^^ not found in `strings`
    |
note: found an item that was configured out
   --> D:\App\Rust\cargo\registry\src\index.crates.io-6f17d22bba15001f\polars-plan-0.37.0\src\dsl\function_expr\strings.rs:433:15
    |
433 | pub(super) fn find(s: &[Series], literal: bool, strict: bool) -> PolarsResult<Series> {
    |               ^^^^
    = note: the item is gated behind the `regex` feature

For more information about this error, try `rustc --explain E0425`.
error: could not compile `polars-plan` (lib) due to previous error
```


### Issue description

Building rust code with feature "strings" prints error, when on polars version 0.37.0.

Building works while back to polars version 0.36.2

### Expected behavior

Building should work on version 0.37.0

### Installed versions

<details>

polars = { version = "0.37.0", features = ["lazy", "strings"] }

</details>

--- a/crates/polars-plan/src/dsl/function_expr/strings.rs
+++ b/crates/polars-plan/src/dsl/function_expr/strings.rs
@@ -47,6 +47,7 @@ pub enum StringFunction {
         dtype: DataType,
         pat: String,
     },
+    #[cfg(feature = "regex")]
     Find {
         literal: bool,
         strict: bool,
@@ -139,6 +140,7 @@ impl StringFunction {
             ExtractGroups { dtype, .. } => mapper.with_dtype(dtype.clone()),
             #[cfg(feature = "string_to_integer")]
             ToInteger { .. } => mapper.with_dtype(DataType::Int64),
+            #[cfg(feature = "regex")]
             Find { .. } => mapper.with_dtype(DataType::UInt32),
             #[cfg(feature = "extract_jsonpath")]
             JsonDecode { dtype, .. } => mapper.with_opt_dtype(dtype.clone()),
@@ -206,6 +208,7 @@ impl Display for StringFunction {
             ExtractGroups { .. } => "extract_groups",
             #[cfg(feature = "string_to_integer")]
             ToInteger { .. } => "to_integer",
+            #[cfg(feature = "regex")]
             Find { .. } => "find",
             #[cfg(feature = "extract_jsonpath")]
             JsonDecode { .. } => "json_decode",
@@ -289,6 +292,7 @@ impl From<StringFunction> for SpecialEq<Arc<dyn SeriesUdf>> {
             ExtractGroups { pat, dtype } => {
                 map!(strings::extract_groups, &pat, &dtype)
             },
+            #[cfg(feature = "regex")]
             Find { literal, strict } => map_as_slice!(strings::find, literal, strict),
             LenBytes => map!(strings::len_bytes),
             LenChars => map!(strings::len_chars),

